### PR TITLE
sesman: remove unnecessary fork for FreeBSD

### PR DESCRIPTION
**Pull request by metalefty**
it is not only unnecessary but causes another bug.

### CODE REVIEW

File: sesman/session.c [hunk 1]
```diff
@@ -513,33 +508,30 @@ session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
...
+        {
+            log_message(LOG_LEVEL_ERROR,
+                        "setlogin failed for user %s - pid %d", s->username,
+                        g_getpid());
         }
-
-        g_waitpid(bsdsespid);
 #endif
+        g_delete_wait_obj(g_term_event);
+        g_tcp_close(g_sck);
```
Comment (proski) [inline sesman/session.c]: Why is this code moved after the FreeBSD specific code? It's better not to hold resources longer than they are needed.

**Comment by metalefty** [comment]
The old code runs session twice due to unnecessary deeper forking. After first session exits, new session starts again.  Besides, the second session doesn't passes FreeBSD specific code.

It's difficult for me to explain what happens by the old code.  Please see the below figure.

```
            this sesman becomes sessvc and forks Xorg, chansrv, session when child sessvc exits
                   |
xrdp-sesman -+- xrdp-sesman --- xrdp-sessvc -+- Xorg        <- first session: setlogined
             +                               +- xrdp-chansrv    this session passes thorough FreeBSD specific patch
             +                               +- xrdp-sesman --- (session manager, ex. Xfce4)
             +
             +
             +- xrdp-sessvc --- Xorg                         <- second session: NOT setlogined
                    |        +- xrdp-chansrv                     this session doesn't passes FreeBSD specific patch, runs after first session exits
                    |        +- xrdp-sesman --- (session manager, ex, Xfce4)   
                    |
              this sessvc was above sesman
```

File: sesman/session.c (hunk 1 above)
Comment (metalefty) [inline sesman/session.c]: Oh, it shouldn't be moved.

**Comment by metalefty** [comment]
The bug is born by myself, found by myself, fixed by myself :pensive: so it is not reported to elsewhere.

**Comment by proski** [comment]
OK then. Just please fix the unnecessarily moved code and let's merge.

By the way, we should split huge functions that do multiple things, fork, wait, loop etc. Bad code structure contributes to such bugs.

Also, it seems strange that xrdp-sesman serves both as a root-owned authenticating daemon and as a user-owned session manager. The later functionality could be split into a separate binary, say, xrdp-session.

**Comment by metalefty** [comment]
I agree with the last sentence. Let's merge later.

**Pull request merged**